tests/Aspire.Cli.Tests/E2E/ExecTests.cs

```diff
@@ -0,0 +1,237 @@
+            "--command", "\"dotnet --help\"",
+            "--postgres"
+        ];
+
+        var app = await BuildAppAsync(args);
+        var logs = await ExecAndCollectLogsAsync(app);
+
+        Assert.True(logs.Count > 0, "No logs were produced during the exec operation.");
+        Assert.Contains(logs, x => x.Text.Contains("Usage: dotnet [sdk-options] [command] [command-options] [arguments]"));
+    }
+
+    [Fact]
+    [RequiresDocker]
+    [QuarantinedTest("https://github.com/dotnet/aspire/issues/10138")]
+    public async Task Exec_InitializeMigrations_ShouldCreateMigrationsInWebApp()
+    {
+        // note: should also install dotnet-ef tool locally\globally
+
+        var migrationName = "AddVersion";
+
+        var apiModelProjectDir = @$"{MSBuildUtils.GetRepoRoot()}\playground\DatabaseMigration\DatabaseMigration.ApiModel\DatabaseMigration.ApiModel.csproj";
+        DeleteMigrations(apiModelProjectDir, migrationName);
+
+        string[] args = [
+            "--operation", "run",
+            "--project", DatabaseMigrationsAppHostProjectPath,
+            "--resource", "api",
+            "--command", $"\"dotnet ef migrations add AddVersion --project {apiModelProjectDir}\"",
+            "--postgres"
+        ];
+
+        var app = await BuildAppAsync(args);
+        var logs = await ExecAndCollectLogsAsync(app, timeoutSec: 60);
+
+        Assert.True(logs.Count > 0, "No logs were produced during the exec operation.");
+        Assert.Contains(logs, x => x.Text.Contains("Build started"));
+        Assert.Contains(logs, x => x.Text.Contains("Build succeeded"));
+
+        AssertMigrationsCreated(apiModelProjectDir, migrationName);
+        DeleteMigrations(apiModelProjectDir, migrationName);
+    }
+
+    private async Task<List<CommandOutput>> ExecAndCollectLogsAsync(DistributedApplication app, int timeoutSec = 30)
+    {
+        var cts = new CancellationTokenSource(TimeSpan.FromSeconds(timeoutSec));
+
+        var appHostRpcTarget = app.Services.GetRequiredService<AppHostRpcTarget>();
+        var outputStream = appHostRpcTarget.ExecAsync(cts.Token);
+
+        var logs = new List<CommandOutput>();
+        var startTask = app.StartAsync(cts.Token);
+        await foreach (var message in outputStream)
+        {
+            var logLevel = message.IsErrorMessage ? LogLevel.Error : LogLevel.Information;
+            var log = $"Received output: #{message.LineNumber} [level={logLevel}] [type={message.Type}] {message.Text}";
+
+            logs.Add(message);
+            output.WriteLine(log);
+        }
+
+        await startTask;
+        return logs;
+    }
+
+    private async Task<DistributedApplication> BuildAppAsync(string[] args, Action<DistributedApplicationOptions, HostApplicationBuilderSettings>? configureBuilder = null)
+    {
+        configureBuilder ??= (appOptions, _) => { };
+        var builder = DistributedApplicationTestingBuilder.Create(args, configureBuilder, typeof(DatabaseMigration_AppHost).Assembly)
```

Comment:
I think for the test cases here I would like to avoid taking a project reference dependency on the database migration project. This might also alleviate the need to make public API changes to the `DistributedApplicationTestingBuilder`. Instead I think a more efficient approach (and a approach that helps better isolate issues) is to build an apphost in the test case itself then instansiate a backchannel to the apphost inside the testcase itself.

Reply:
> Instead I think a more efficient approach (and a approach that helps better isolate issues) is to build an apphost in the test case itself then instansiate a backchannel to the apphost inside the testcase itself.

Sorry, did not catch this idea. I answered about the approach taken here: https://github.com/dotnet/aspire/pull/10240#discussion_r2184830793